### PR TITLE
Added abstraction for OS commands.

### PR DESCRIPTION
**Pull request by ashcrow**
Adds an abstraction layer for getting the right commands for different OS types.

### CODE REVIEW

File: src/commissaire/oscmd/fedora.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    os_type = 'fedora'
+
+    def reboot(self):
+        """
+        Fedora reboot command.
+
+        :return: The command to execute as a list
+        :rtype: list
+        """
+        return ['reboot']
```
Comment (mbarnes) [inline src/commissaire/oscmd/fedora.py]: Any reason why not `['systemctl', 'reboot']` everywhere?

Comment (ashcrow) [inline src/commissaire/oscmd/fedora.py]: No specific reason. I'll update it.

File: src/commissaire/oscmd/rhel.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://
+"""
+RHEL commands.
+"""
+
+from commissaire.oscmd import OSCmdBase
+
+
+class OSCmd(OSCmdBase):
```
Comment (mbarnes) [inline src/commissaire/oscmd/rhel.py]: If the `oscmd` classes will stay this simple it doesn't really matter, but you could inherit from the Fedora class here and just tweak `os_type` (like how RHEL and Fedora actually work).  Would reduce some duplication.

File: src/commissaire/oscmd/rhel.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    """
+    Commmands for RHEL.
+    """
+
+    #: The type of Operating System
+    os_type = 'rhel'
+
+    def reboot(self):
+        """
+        Fedora reboot command.
```
Comment (mbarnes) [inline src/commissaire/oscmd/rhel.py]: s/Fedora/RHEL/ ?

File: src/commissaire/oscmd/rhel.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+        :return: The command to execute as a list
+        :rtype: list
+        """
+        return ['reboot']
+
+    def upgrade(self):
+        """
+        Fedora upgrade command.
```
Comment (mbarnes) [inline src/commissaire/oscmd/rhel.py]: s/Fedora/RHEL/ ?

**Comment by mbarnes** [comment]
Looks good overall.  :+1:

I kinda feel like it might read better if the subpackage was just named `os`.   Then you could maybe drop the `OS` part from the class names.  I don't know, `oscmd` just struck me as kinda awkward.  But not a big deal and totally optional.

**Comment by ashcrow** [comment]
@mbarnes ```oscmd``` is a little awkward. I'd like to avoid calling it ```os``` though since there is a Python stdlib with that name. I'm open to other names though (keep in mind ```os```, ```platform```, ```cmd```, and ```sys``` are all stdlib).

**Comment by mbarnes** [comment]
Ah, I didn't think that would be an issue since it's under the `commissaire` namespace.  But maybe it's bad practice anyway.

Fixup looks good.  I'll merge.

**Comment by ashcrow** [comment]
@mbarnes It wouldn't be an issue but it would bug me since one could import os from ```commissaire``` or from stdlib. Both would have the same name yet provide different functionality. More a personal preference thing than anything else.

**Comment by mbarnes** [comment]
Fair enough.  Squashed and merged.

**Pull request closed without merging**